src/Units/Unit.java

```diff
@@ -191,79 +191,59 @@ public void fire(WeaponModel weapon)
     }
   }
 
-  /** @return unit HP * 10 */
-  public int getEffectiveHealth()
-  {
-    return effectiveHealth(health);
-  }
-  /** @return unit actual health */
-  public int getPreciseHealth()
-  {
-    return health;
-  }
   public boolean isHurt()
   {
     return health < healthFromHP(model.maxHP);
   }
   public int getHP()
   {
-    return healthToHP(health);
+    return (int) Math.ceil(healthToHP(health));
   }
   /** @return value in range [0-1.0]; represents the unit's current effectiveness */
   public double getHPFactor()
   {
     return getHP() / (double)model.maxHP;
   }
-  /** @return health value == its corresponding HP value * 10 */
-  public static int effectiveHealth(int input)
+  /** @return un-rounded HP */
+  public double getPreciseHP()
   {
-    return (input - input%10) + ((input%10 > 0)? 10:0);
+    return healthToHP(health);
   }
-  public static int healthToHP(int input)
+  private static double healthToHP(int input)
   {
-    return effectiveHealth(input) / 10;
+    return ((double)input)/10;
   }
-  public static int healthFromHP(int input)
+  private static int healthFromHP(double input)
   {
-    return input * 10;
+    return (int) (input * 10);
```

Comment:
You mentioned wanting to leave open the possibility of units with `maxHP != 10`. Shouldn't this 10 and the one above be a `model.maxHP` reference? I know these are static right now, but maybe they shouldn't be, unless we'd rather pass in the model.

Reply:
Nope. This is a 100% static conversion factor - it defines 1 HP == 10 health, irrespective of how much HP you *can* have.
So if I have 110 health and I divided by maxHP, I would report 10 HP instead of 11 HP.